File: solutions/iq-test.py
def iq_test(numbers):
    numbers = list(map(int, numbers.split(" ")))
    first_three = numbers[:3]
    evens = len([ n for n in first_three if n % 2 == 0])
    odds = len([ n for n in first_three if n % 2 == 1])

    find = 1
    if evens < odds:
        find = 0

    for i, n in enumerate(numbers, 1):
        if find == n % 2:
            return i

# iq_test("2 4 7 8 10") => 3 # Third number is odd, while the rest of the numbers are even
# iq_test("1 2 1 1") => 2 # Second number is even, while the rest of the numbers are odd

File: solutions/iq-test.js
function iqTest(numbers){
    numbers = numbers.split(' ').map(x => parseInt(x))
    let evens = 0, odds = 0
    let last_odd = 0, last_even = 0
    for (let i = 0, n = 0; i < numbers.length; i++) {
      n = numbers[i]
      if ((n % 2 == 0)){
        if (odds > 1) return i + 1
        last_even = i
        evens += 1
      }
      else if ((n % 2 == 1)){
        if (evens > 1) return i + 1
        odds += 1
        last_odd = i
      }
    }
    if (evens > odds) return last_odd + 1
    else return last_even + 1
  }
  